miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -622,37 +606,73 @@ end
 # ACCOUNT DATA
 # -------------------------------------------------------------------------------------------------
 
-#! Returns a pointer to the account data.
+#! Sets the offset of the current account data to the native account (2048).
+#!
+#! Stack: []
+#! Output: [native_account_data_ptr]
+#!
+#! Where:
+#! - native_account_data_ptr is the memory address at which the native account data is stored.
+export.set_current_account_data_offset_to_native_account
+    push.NATIVE_ACCOUNT_DATA_PTR
+    push.CURRENT_ACCOUNT_DATA_OFFSET mem_store
+end
+
+#! Returns the offset of the current account data.
 #!
 #! Stack: []
 #! Output: [ptr]
 #!
 #! Where:
 #! - ptr is the memory address at which the account data begins.
-export.get_acct_data_ptr
-    push.ACCT_DATA_SECTION_OFFSET
+export.get_current_account_data_offset
+    push.CURRENT_ACCOUNT_DATA_OFFSET mem_load
+end
+
+#! Sets the offset of the current account data to the provided value.
+#!
+#! Stack: [curr_account_data_offset]
+#! Output: []
+#!
+#! Where:
+#! - curr_account_data_offset is the data offset of the currently accessing account.
+export.set_current_account_data_offset
+    push.CURRENT_ACCOUNT_DATA_OFFSET mem_store
 end
 
-#! Returns a pointer to the end of the account data section.
+#! Returns a pointer to the end of the core account data section.
 #!
 #! Stack: []
 #! Output: [ptr]
 #!
 #! Where:
-#! - ptr is the memory address at which the account data ends.
+#! - ptr is the memory address at which the core account data ends.
 export.get_core_acct_data_end_ptr
-    push.ACCT_CORE_DATA_SECTION_END_OFFSET
+    exec.get_current_account_data_offset push.ACCT_CORE_DATA_SECTION_END_OFFSET add
 end
 
-#! Returns the account id.
+#! Returns the id of the current account.
 #!
 #! Stack: []
-#! Output: [acct_id]
+#! Output: [curr_acct_id]
 #!
 #! Where:
-#! - acct_id is the account id.
-export.get_acct_id
-    push.ACCT_ID_AND_NONCE_PTR mem_load
+#! - curr_acct_id is the account id of the currently accessing account.
+export.get_current_account_id
```

Comment:
I would probably keep this as it was. Otherwise, we should rename most other methods here to be consistent (e.g., `get_current_account_nonce`).